components/table/table-wrapper.js

```diff
@@ -575,12 +576,27 @@ export class TableWrapper extends RtlMixin(PageableMixin(SelectionMixin(LitEleme
 	}
 
 	async _handleTableChange() {
-		await new Promise(resolve => requestAnimationFrame(resolve));
+		const debounceDelay = 300;
+		const update = async() => {
+			await new Promise(resolve => requestAnimationFrame(resolve));
 
-		this._updateItemShowingCount();
-		this._applyClassNames();
-		this._syncColumnWidths();
-		this._updateStickyTops();
+			this._updateItemShowingCount();
+			this._applyClassNames();
+			this._syncColumnWidths();
+			this._updateStickyTops();
+			this.dispatchEvent(new CustomEvent('d2l-table-changed', { bubbles: true, composed: false }));
+			this._recentTableChange = true;
+			this._recentTableChange = setTimeout(() => this._recentTableChange = false, debounceDelay);
+		};
+
+		if (!this._recentTableChange) {
+			update();
+		} else if (!this._tableChangeTimeout) {
+			this._tableChangeTimeout = setTimeout(() => {
+				this._tableChangeTimeout = null;
+				update();
+			}, debounceDelay);
+		}
```

Comment:
The pattern you're using is more of a "throttle" than a "debounce"? I think a typical debounce pattern would be to cancel any previous timeouts that are in progress and then trigger a new one, which would mean that the code running in the timeout is always the "latest" one.

It would however mean that if changes happened constantly over and over again indefinitely, the timeout code would never actually run, instead of just running once every 300ms. Not sure what's best in this scenario.

I think would simplify your code though. Something like:

```javascript
if (!this._recentTableChange) {
  update();
} else {
  clearTimeout(this._tableChangeTimeout);
  this._tableChangeTimeout = setTimeout(update, debounceDelay);
}
```

Reply:
I'm not sure what the best is either. Originally Gianfranco had it configured to wait until 300ms without any changes and I noted that, in theory, it could wait indefinitely. I think Gianfranco made a couple changes as a result.

Whichever way we go, I think we should ensure that it still runs once immediately before beginning the debouncing/throttling.